Diarias

RuanMoura opened the pull request:
Implementa a duas funções de get uma para a quantidade de diárias diurnas e outra para a quantidade de diárias noturnas dado um horários de acesso e um estacionamento.

DaviAntonio committed:
Validar horários noturnos ao criar estacionamento

Os testes de criação foram alterados para cobrir o caso da validação dos
horários noturnos no ato da criação do objeto de Estacionamento.

O valor de entrada norturno deve ser posterior ao de saída, por se
tratar de um horário do dia anterior.

DaviAntonio committed:
Validar horários noturnos ao criar estacionamento

Os testes de criação foram alterados para cobrir o caso da validação dos
horários noturnos no ato da criação do objeto de Estacionamento.

O valor de entrada norturno deve ser posterior ao de saída, por se
tratar de um horário do dia anterior.

DaviAntonio committed:
Validar horários de abertura e encerramento

O horário de abertura deve ser anterior ao de encerramento, por
acontecerem no mesmo dia.

DaviAntonio committed:
Validar horários de abertura e encerramento

O horário de abertura deve ser anterior ao de encerramento, por
acontecerem no mesmo dia.

DaviAntonio committed:
Adicionar testes de horários inválidos às suítes

DaviAntonio committed:
Adicionar testes de horários inválidos às suítes

Commit:
Merge branch 'registra_estacionamento' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into registra_estacionamento

GuilhermeBraz committed:
parametrizar RegistraHorarioEntradaDiariaNoturnaTest

GuilhermeBraz committed:
parametrizar RegistraHorarioSaidaDiariaNoturnaTest

GuilhermeBraz committed:
parametrizar RegistraHorarioAberturaTest

GuilhermeBraz committed:
corrigir nome teste

GuilhermeBraz committed:
parametrizar RegistraHorarioEncerramentoTest

GuilhermeBraz committed:
parametrizar RegistraCapacidadeTest -falta teste excecao

GuilhermeBraz committed:
parametrizar RegistraIdEstacionamentoTest -falta teste excecao

GuilhermeBraz committed:
parametrizar RegistraValorDiariaDiurnaEstacionamentoTest -falta teste excecao

RuanMoura committed:
parametrizar RegistraValorDiariaNoturnaEstacionamentoTest

DaviAntonio committed:
Validar criação de estacionamento sem id

RuanMoura committed:
parametrizar RegistraValorEventoEstacionamentoTest

RuanMoura committed:
parametrizar RegistraValorFracaoEstacionamentoTest

RuanMoura committed:
parametrizar RegistraValorHoraCheiaEstacionamentoTest

RuanMoura committed:
parametrizar RegistraValorMensalistaEstacionamentoTest

RuanMoura committed:
parametrizar RegistraValorRetornoContratanteEstacionamentoTest

DaviAntonio committed:
Validar criação de estacionamentos parciais

DaviAntonio committed:
Adicionar teste de estacionamentos parcias às suítes

DaviAntonio committed:
Merge branch 'registra_estacionamento' of https://github.com/TPPE-2022-1/TP-1-Test-Driven-Development into registra_estacionamento

RuanMoura committed:
parametrizar CriaEstacionamentoTest

RuanMoura committed:
Merge branch 'registra_estacionamento' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into registra_estacionamento

DaviAntonio committed:
Adicionar testes faltantes às suítes

CriaAcessoTest, RegistroHoraEntradaTest, RegistroHoraSaidaTest, e
RegistroPlacaTest foram adicionados às suítes de exceções e de
funcionalidades testadas.

Commit:
Falsificação - Calculo de Frações - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Falsificação - Calculo de Frações - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Duplicação - Calculo de Frações - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Duplicação - Calculo de Frações - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Triangulação - Calculo de Frações - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Triangulação - Calculo de Frações - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Parametrização - Calculo de Fracoes - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Parametrização - Calculo de Fracoes - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Falsificação - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Falsificação - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Duplicação - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Duplicação - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Triangulação - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Triangulação - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Parametrização - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Parametrização - Calculo de Hora Cheia - Teste Funcional

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

ilus1 commented on src/main/java/tppe/tp1/acesso/AcessoBuilder.java:
```diff
@@ -43,6 +44,29 @@ public Float getValorAcesso() {
 		return valorContratante;
 	}
+	
+	public Integer getDiariasDiurnas(EstacionamentoBuilder e) {
```
Se for pra retornar 0 ou  1 é melhor usar boolean

Se for usar boolean poe o nome da função de isDiariaDurna, pra ja passar a ideia de checagem true/false

RuanMoura replied:
Feito :+1:

ilus1 commented on src/main/java/tppe/tp1/acesso/AcessoBuilder.java:
```diff
@@ -43,6 +44,29 @@ public Float getValorAcesso() {
+	public Integer getDiariasDiurnas(EstacionamentoBuilder e) {
+		LocalTime entrada = getHoraEntrada();
+		LocalTime saida = getHoraSaida();
+		if(e.getHorarioSaidaDiariaNoturna().compareTo(entrada) > 0) {
```
Se eu não me engano a checagem seria if(e.getHorarioEntradaDiariaNoturna().compareTo(entrada) > 0)

Já que pra contar como diaria noturna o carro precisa entrar depois do horario marcado como horarioEntradaDiariaNoturna e sair antes do horarioSaidaDiariaNoturna

mas eu posso estar enganado 🤔

ilus1 commented on src/main/java/tppe/tp1/acesso/AcessoBuilder.java:
```diff
@@ -43,6 +44,29 @@ public Float getValorAcesso() {
+		if(e.getHorarioEntradaDiariaNoturna().compareTo(saida) < 0) {
+			saida = e.getHorarioEntradaDiariaNoturna();
+		}
+		if(saida.toSecondOfDay() - entrada.toSecondOfDay() < 32400)
```
Tem q dar uma checada, mas eu acho que aqui seria 31560

Commit:
Teste do registro dos tipos de acesso

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste do registro dos tipos de acesso

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de entrada - Falsificação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de entrada - Falsificação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de Acesso - Duplicação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de Acesso - Duplicação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de acesso - Triangulação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de acesso - Triangulação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de Acesso - Parametrização

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de Acesso - Parametrização

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de acessoEvento - falsificação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de acessoEvento - falsificação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de AcessoEvento - Duplicação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de AcessoEvento - Duplicação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de AcessoEvento - Tiangulação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de AcessoEvento - Tiangulação

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de AcessoEvento - Parametrização

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

Commit:
Teste Funcional - Checagem de tipo de AcessoEvento - Parametrização

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>

RuanMoura committed:
Falsificação - Diaria Diurna

RuanMoura committed:
Falsificação - Diaria Diurna

RuanMoura committed:
Duplicação - Diaria Diurna

RuanMoura committed:
Duplicação - Diaria Diurna

RuanMoura committed:
Triangulação - Diaria Diurna

RuanMoura committed:
Triangulação - Diaria Diurna

RuanMoura committed:
Falsificação - Diaria Noturna

RuanMoura committed:
Falsificação - Diaria Noturna

RuanMoura committed:
Duplicação - Diaria Noturna

RuanMoura committed:
Duplicação - Diaria Noturna

RuanMoura committed:
Triangulação - Diaria Noturna

RuanMoura committed:
Triangulação - Diaria Noturna

GuilhermeBraz committed:
adicionar verificacao para valor nulo

GuilhermeBraz committed:
adicionar verificacao para valor nulo

GuilhermeBraz committed:
adicionar verificacao para valor nulo

ilus1 committed:
Merge pull request #4 from TPPE-2022-1/registra_estacionamento

Que deus nos abençoe

Commit:
Merge branch 'Calculo_Fracao_Hora' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into Calculo_Fracao_Hora

RuanMoura committed:
mudança de tempo necessario para ser considerado uma diaria

RuanMoura committed:
mudança de tempo necessario para ser considerado uma diaria

Commit:
Adição dos testes na suite

Commit:
Correção no calculo de frações

Commit:
Parametrização do teste de tipo de acesso

GuilhermeBraz committed:
Merge pull request #5 from TPPE-2022-1/Calculo_Fracao_Hora

Calculo fracao hora

GuilhermeBraz committed:
Adicionar acesso em lista de acessos

GuilhermeBraz committed:
Adicionar teste CadastraAcessoEmListaEstacionamentoTest na suite de testes

GuilhermeBraz committed:
Adicionado teste com retorno da lista inteira

DaviAntonio committed:
Corrigir verificação de lista de acesso vazia

A verificação de lista de acesso vazia deve validar se a lista não
possui elementos, e não se ela não é nula, pois o objeto de lista é
criado automaticamente ao se criar o objeto de estacionamento.

DaviAntonio committed:
Verificar métodos que não deveriam lançar exceções no teste da lista

ilus1 committed:
Merge pull request #6 from TPPE-2022-1/registra_estacionamento

Registra estacionamento

Commit:
Correção de conflitos

DaviAntonio committed:
Inicia Migração do registro de entrada para incluir data

Commit:
Inicio da aplicação

Commit:
Merge branch 'diaria' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into diaria

GuilhermeBraz committed:
Adicionar funcao verificar todas as listas no menu

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>
Co-Authored-by: Fernando Miranda <ilusdasilva@gmail.com>

GuilhermeBraz committed:
Trocar tipo das variaveis de Float para Double

Co-authored-by: antonioaldisio <aldisiofilho@gmail.com>
Co-Authored-by: Fernando Miranda <ilusdasilva@gmail.com>

Commit:
Remoção de imports nao usados

Commit:
Merge branch 'diaria' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into diaria

DaviAntonio committed:
Corrigir nome da classe principal

AntonioAldisio committed:
Teste funcional - calculo do valor de acesso - falsificacao

Co-Authored-By: Fernando Miranda <ilusdasilva@gmail.com>
Co-Authored-By: Guilherme Braz <44117439+GuilhermeBraz@users.noreply.github.com>

AntonioAldisio committed:
Merge branch 'diaria' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into diaria

AntonioAldisio committed:
Teste funcional - calculo do valor de acesso - duplicacao

Co-Authored-By: Fernando Miranda <ilusdasilva@gmail.com>
Co-Authored-By: Guilherme Braz <44117439+GuilhermeBraz@users.noreply.github.com>

Commit:
Correção no calculo da DiariaNoturna

RuanMoura committed:
corrige diarais diurnas

Commit:
Teste Funcional - Calculo do valor de Acesso - Falsificação

Commit:
Adicionando testes a suite

Commit:
Teste Funcional - Calculo do valor de Acesso - Duplicação

Commit:
Remoção de argumento desnecessário na checagem de tipo

Commit:
Teste Funcional - Calculo do valor e acesso - Triangulação

Commit:
Teste Funcional - Calculo Valor do Acesso - Parametrização

Commit:
Teste Funcional - Calculo Valor Cotnratante - Falsificação

Commit:
Teste Funcional - Calculo do valor do Contratante - Duplicação

GuilhermeBraz committed:
calculo acesso

Commit:
Teste Funcional - Calculo de Valor contratante - Triangulação

Commit:
Merge branch 'diaria' of github.com:TPPE-2022-1/TP-1-Test-Driven-Development into diaria

DaviAntonio reviewed (approved):
Testes OK

Commit:
Parametrização dos testes de calculo de valor de contratante

GuilhermeBraz reviewed (approved):
Pr ok

AntonioAldisio commented:
Suite de Teste ok

DaviAntonio reviewed (approved):
Suites OK, Merge prosseguirá

ilus1 commented:
Achei meio bosta :+1:
